### PR TITLE
cross compile grpc-rs to windows under *nix

### PR DESCRIPTION
**Pull request by light4**
仅作参考

### CODE REVIEW

**Comment by BusyJay** [comment]
Thanks for your contribution! Can you translate the documentation into English?

File: cross_compile.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Mac 下交叉编译 windows 版本 grpc-rs
```
Comment (siddontang) [inline cross_compile.md]: please use English comments

**Comment by siddontang** [comment]
PTAL @queenypingcap

**Comment by queenypingcap** [comment]
@lilin90 PTAL

File: cross_compile.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            .define("CMAKE_SYSTEM_PROCESSOR", get_env("CARGO_CFG_TARGET_ARCH").unwrap())
+            .define("CMAKE_CROSSCOMPILING", "true")
+            .build_target(library)
+            .uses_cxx11()
+            .build()
+        // config.build_target(library).uses_cxx11().build()
+    };
+```
+
+#### fix find zlib
```
Comment (lilin90) [inline cross_compile.md]: "#### fix" -> "### Fix"

or **"## Fix find zlib"** if it is the same level with "Fix CMake"

File: cross_compile.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+```rust
+    // try these values
+    let mut zlib = "z";
+    let mut zlib = "zlibstatic";
+    let mut zlib = "zlibstaticd";
+```
+
+### fix try_run
```
Comment (lilin90) [inline cross_compile.md]: "### fix" -> "## Fix"

File: cross_compile.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+SET( RUN_HAVE_STD_REGEX
+     0
+     CACHE STRING "Result from TRY_RUN" FORCE)
+
+SET( RUN_HAVE_STEADY_CLOCK
+     0
+     CACHE STRING "Result from TRY_RUN" FORCE)
+```
+
+### fix WIN32 API
```
Comment (lilin90) [inline cross_compile.md]: "### fix" -> "## Fix"

File: cross_compile.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+```
+# grpc-rs/grpc-sys/grpc/CMakeLists.txt
+# add these code after about line number 295
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -D_WIN32_WINNT=0x600")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D_WIN32_WINNT=0x600")
+set(C_CXX_FLAGS "${C_CXX_FLAGS} -D_WIN32_WINNT=0x600")
+```
+
+### fix boringssl
```
Comment (lilin90) [inline cross_compile.md]: "### fix" -> "## Fix"

File: cross_compile.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
+# Cross Compile gRPC-rs(0.2.1) to Windows under *nix
+
+## Fisrt you need to install mingw
```
Comment (lilin90) [inline cross_compile.md]: Spelling error: Fisrt -> First

**Comment by Hoverbear** [comment]
resolved conflicts, @ice1000 PTAL again

**Pull request merged**